source/connection-string/connection-string-spec.md

```diff
@@ -216,13 +216,24 @@ The values in connection options MUST be URL decoded by the parser. The values c
   ```
 
 - Key value pairs: A value that represents one or more key and value pairs. Multiple key value pairs are delimited by a
-  comma (","). The key is everything up to the first colon sign (":") and the value is everything afterwards. If any
-  keys or values containing a comma (",") or a colon (":") they must be URL encoded. For example:
+  comma (","). The key is everything up to the first colon sign (":") and the value is everything afterwards. Reserved
+  characters such as ':' must be escaped according to [RFC 2396](https://www.rfc-editor.org/rfc/rfc2396)
+
+  For example:
 
   ```
   ?readPreferenceTags=dc:ny,rack:1
   ```
 
+  Drivers MUST handle unencoded colon signs (":") within the value. For example:
+
+  ```
+  ?authMechanismProperties=TOKEN_RESOURCE:mongodb%3A%2F%2F...
```

Comment:
Wouldn't "unencoded" be:
```suggestion
  ?authMechanismProperties=TOKEN_RESOURCE:mongodb://...
```
But, I think this should just be a test. In general, drivers should handle all unreserved characters that are unencoded, in any context, just because the spec does not reserve them, and because percent-decoding does not alter them. This appears to be an existing consequence of how the spec is written.

Reply:
Clarified